Edits from Fedora package review

kalebskeithley opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/linux-ha-storage/storhaug/9)
<!-- Reviewable:end -->

kalebskeithley committed:
Edits from Fedora package review

kalebskeithley committed:
setup_cluster needs to wait for certs to propagate and cluster to elect a DC

jarrpa commented on src/storhaug:
```diff
@@ -324,15 +324,28 @@ setup_cluster()
+
+    # BZ 1284404, 1425110, allow time for SSL certs to propagate, until then
+    # pcsd will not accept connections.
+    sleep 12
```
Given the loop below, this seems unnecessary.

kalebskeithley replied:
this is what the pacemaker devs told me to do.

jarrpa commented on src/storhaug:
```diff
@@ -324,15 +324,28 @@ setup_cluster()
+    while [ $? -ne 0 ]; do
+        sleep 2
+        pcs cluster start --all || storlog "ERR" "Failed to start cluster ${HA_NAME}"
+    done
```
This loop has no exit condition, couldn't it hang forever?

kalebskeithley replied:
seems not to.

jarrpa replied:
But what happens if the command never succeeds? What if the cluster barfs mid-startup?

jarrpa commented on src/storhaug:
```diff
@@ -324,15 +324,28 @@ setup_cluster()
+    crmadmin --dc_lookup --timeout=5000 > /dev/null 2>&1
+    while [ $? -ne 0 ]; do
+        crmadmin --dc_lookup --timeout=5000 > /dev/null 2>&1
+    done
```
This loop has no exit condition, couldn't it hang forever?

kalebskeithley replied:
see above

jarrpa commented on src/storhaug:
```diff
@@ -324,15 +324,28 @@ setup_cluster()
     pcs cluster auth ${HA_SERVERS} -u hacluster -p ${HA_PASSWORD} --force
     pcs cluster setup --force --name ${HA_NAME} ${HA_SERVERS} || storlog "ERR" "Failed to setup cluster ${HA_NAME}"
+
+    # BZ 1284404, 1425110, allow time for SSL certs to propagate, until then
```
Don't reference BZ numbers unless you're specifying what bugzilla system they belong to.

jarrpa commented on src/storhaug:
```diff
@@ -324,15 +324,28 @@ setup_cluster()
+    done
+
+    # wait for the cluster to elect a DC before querying or writing
+    # to the CIB. BZ 1334092
```
Don't reference BZ numbers unless you're specifying what bugzilla system they belong to.

jarrpa commented on scripts/storhaug.spec:
```diff
@@ -52,82 +47,41 @@ Requires: samba-winbind-clients
 %config(noreplace) %{_sysconfdir}/sysconfig/storhaug.d/nfs-ha.conf
-%{_prefix}/lib/ocf/resource.d/heartbeat/ganesha
-%{_prefix}/lib/ocf/resource.d/heartbeat/ganesha_trigger
+%{_prefix}/lib/ocf/resource.d/heartbeat/*
```
This seems wrong... wouldn't this catch all heartbeat RAs?

kalebskeithley replied:
there are only thos two in rpm BUILDROOT tree, so no.
And that's what the Fedora provenpackager reviewer wanted, so also no

jarrpa replied:
Ah! All righty then. :)

kalebskeithley committed:
gluster shared storage vol is gluster_shared_storage

kalebskeithley committed:
more cleanups for storhaug for imminent glusterfs-3.11

use systemctl to start nfs-ganesha/ganesha.nfsd. No, this doesn't
work on RHEL6

mostly consistent use of ${foo} versus $foo everywhere

fixup several incorrect path or path components for gluster shared
storage

jarrpa commented on src/nfs-ha.conf.sample:
```diff
@@ -2,7 +2,7 @@
 
 # Shared volume to store NFS state.
-HA_NFS_VOL="state"
+HA_NFS_VOL="gluster_shared_storage"
```
Change to just `shared_storage`

jarrpa commented on src/storhaug:
```diff
@@ -54,20 +56,32 @@ usage()
     case $(eval echo \${${1}}) in
-        TRUE | True | true | YES | Yes | yes) declare "${1}"=true ;;
-        FALSE | False | false | NO | No | no) declare "${1}"=false ;;
+        [Tt][Rr][Uu][Ee] | [Yy][Ee][Ss]) declare "${1}"=true ;;
```
Smart. :)

jarrpa commented on src/storhaug:
```diff
@@ -291,29 +305,44 @@ create_virt_ip()
-            flush_routes="true" \
-        op monitor interval=60s \
-        meta resource-stickiness="0"
+    if [[ "X${HA_SVC_NFS}X" == "XyesX" ]]; then
```
There should only be one definition of the vipN resource, prior to this block, automatically part of a group (even if it's a group of only itself). This will remove the need for the `else` statement below and make it easier to add portblock for SMB.

jarrpa commented on src/storhaug:
```diff
@@ -291,29 +305,44 @@ create_virt_ip()
-        meta resource-stickiness="0"
+    if [[ "X${HA_SVC_NFS}X" == "XyesX" ]]; then
+        pcs -f ${cibfile} resource create vip${ipcount} ocf:heartbeat:IPaddr2 \
+            --group group${ipcount} \
```
Make the group name something like `vip_groupN` instead.

jarrpa commented on src/storhaug:
```diff
@@ -291,29 +305,44 @@ create_virt_ip()
+            op monitor interval=60s \
+            meta resource-stickiness="0"
+
+        pcs -f ${cibfile} resource create vip${ipcount}_block ocf:heartbeat:portblock \
```
Rename to `vipN_nfs_block`

jarrpa commented on src/storhaug:
```diff
@@ -291,29 +305,44 @@ create_virt_ip()
+        pcs -f ${cibfile} resource create vip${ipcount}_block ocf:heartbeat:portblock \
+            protocol=tcp portno=2049 action=block ip=${ip} --group group${ipcount}
+
+        pcs -f ${cibfile} resource create vip${ipcount}_unblock ocf:heartbeat:portblock \
```
Rename to `vipN_nfs_unblock`

jarrpa commented on src/storhaug:
```diff
@@ -291,29 +305,44 @@ create_virt_ip()
-    pcs -f ${cibfile} constraint colocation add vip${ipcount}_trigger-clone with vip${ipcount} INFINITY
-    pcs -f ${cibfile} constraint order vip${ipcount} then vip${ipcount}_trigger-clone
-
-    pcs -f ${cibfile} constraint colocation add vip${ipcount} with nfs-ganesha-clone INFINITY
```
Why are you getting rid of the service constraints? A vip shouldn't start on a node that doesn't have the configured services running.

jarrpa commented on src/storhaug:
```diff
@@ -364,67 +393,71 @@ setup_cluster()
+        pcs -f ${cibfile} resource create nfs-ganesha ocf:heartbeat:ganesha \
+            params \
+                config="${HA_NFS_CONF}" \
             state_fs="${HA_NFS_TYPE}" \
```
This should be further indented.

jarrpa reviewed (changes requested):
There's a lot of things going on here. I'd like to see this commit split into at least three or four commits.
